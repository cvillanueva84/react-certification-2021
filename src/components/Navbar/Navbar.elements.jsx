import React from 'react';
import styled from 'styled-components';
import { AiFillYoutube } from 'react-icons/ai';
import { FaSearch } from 'react-icons/fa';

export const Nav = styled.nav`
  background: ${(props) => props.theme.background_color};
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100vw;
  height: ${(props) => props.theme.navbar_height};
  position: fixed;
  top: 0;
  z-index: 99;
  padding: 0rem 1.5rem;
`;

const LogoContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: flex-start;
  align-items: center;
  user-select: none;
`;

const LogoText = styled.h1`
  color: ${(props) => props.theme.text_color};
  margin-left: 0.5rem;
  font-size: 2rem;
  font-weight: 900;
  @media (max-width: 600px) {
    display: none;
  }
`;

export function IconLogo() {
  return (
    <LogoContainer>
      <AiFillYoutube size="3.5rem" color="red" />
      <LogoText>WizeTube</LogoText>
    </LogoContainer>
  );
}

const FormSearch = styled.form`
  display: flex;
  display: row;
`;

const InputSearch = styled.input`
  padding: 0rem 0.5rem;
  border: 1px solid ${(props) => props.theme.icon_color};
  margin-left: 1rem;
  width: 350px;

  @media (max-width: 737px) {
    width: 200px;
  }
`;

const SearchButton = styled.button`
  color: ${(props) => props.theme.icon_color};
  display: flex;
  justify-content: center;
  align-content: center;
  padding: 0.5rem 0.75rem;
  margin-right: 1rem;
`;

export function SearchBar() {
  return (
    <FormSearch>
      <InputSearch type="text" placeholder="Search" />
      <SearchButton>
        <FaSearch size="1rem" />
      </SearchButton>
    </FormSearch>
  );
}

export const ProfileImg = styled.img`
  border-radius: 50%;
  width: 3.5rem;
  height: auto;
`;
