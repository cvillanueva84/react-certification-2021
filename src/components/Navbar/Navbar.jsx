import React, { useState } from 'react';
import { GiHamburgerMenu } from 'react-icons/gi';
import { BiSearchAlt2 } from 'react-icons/bi';
import {
  Nav,
  Menu,
  Input,
  ContainerInput,
  InputCheck,
  Image,
  Menu2,
  ButtonMenu,
  NavMenu,
  NavItem,
  DivItem,
  SpanItem,
} from './Navbar.styled';

const Navbar = () => {
  const [click, setClick] = useState(false);
  const handleClick = () => setClick(!click);

  return (
    <>
      <Nav>
        <Menu>
          <NavMenu onClick={handleClick} click={click}>
            <NavItem>
              <DivItem>
                <SpanItem>Home</SpanItem>
              </DivItem>
            </NavItem>
          </NavMenu>
          <ButtonMenu onClick={handleClick}>
            <GiHamburgerMenu />
          </ButtonMenu>
          <ContainerInput>
            <BiSearchAlt2 style={{ marginRight: '8px' }} />
            <Input placeholder="Search..." />
          </ContainerInput>
        </Menu>
        <Menu2>
          <InputCheck type="checkbox" />
          <span>Dark mode</span>
          <Image>
            <img
              src="https://w7.pngwing.com/pngs/340/956/png-transparent-profile-user-icon-computer-icons-user-profile-head-ico-miscellaneous-black-desktop-wallpaper.png"
              alt="Fotografia"
            />
          </Image>
        </Menu2>
      </Nav>
    </>
  );
};
export default Navbar;
