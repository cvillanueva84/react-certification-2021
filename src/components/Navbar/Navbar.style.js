import { makeStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme) => ({
    grow: {
        flexGrow: 1,
    },
    title: {
        display: 'none',
        [theme.breakpoints.up('sm')]: {
          display: 'block',
        },
    },
    sectionDesktop: {
        display: 'none',
        [theme.breakpoints.up('md')]: {
          display: 'flex',
        },
    },
    sectionMobile: {
        display: 'flex',
        [theme.breakpoints.up('md')]: {
            display: 'none',
        },
    }
}));
