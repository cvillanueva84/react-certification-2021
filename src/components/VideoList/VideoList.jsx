import React, { useEffect, useState } from 'react';
import axios from 'axios';
import { DivTitle, Container, Fakegrid } from './VideoList.styled';
import Card from '../Card/Card';

const baseURL =
  'https://gist.githubusercontent.com/jparciga/1d4dd34fb06ba74237f8966e2e777ff5/raw/f3af25f1505deb67e2cc9ee625a633f24d8983ff/youtube-videos-mock.json';

const VideoList = () => {
  const [videos, setVideos] = useState([]);

  useEffect(() => {
    async function getVideos() {
      const {
        data: { items },
      } = await axios.get(baseURL);
      const aux = [...items];
      aux.shift();
      setVideos(aux);
      // console.log(data.items.map((el) => el.snippet.description));
    }
    getVideos();
  }, []);

  return (
    <>
      <DivTitle>
        <h2>Welcome to the Challenge!</h2>
      </DivTitle>
      <Container>
        <Fakegrid>
          {videos.map((video) => (
            <Card
              title={video.snippet.title}
              description={video.snippet.description}
              link={video.snippet.thumbnails.high.url}
              key={video.snippet.publishedAt}
            />
          ))}
        </Fakegrid>
      </Container>
    </>
  );
};

export default VideoList;
