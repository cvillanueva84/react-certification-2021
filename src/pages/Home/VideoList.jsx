import React from 'react';
import styled from 'styled-components';
import VideoItem from './VideoItem';

const VideoListContainer = styled.div`
    width: 100%;
    height: 100vh;
    margin: 5px auto;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    
    & a {
        text-decoration: none;
    }
`;

const VideoList = ({videos}) => {

    if (videos.length < 0 ) return null;

    return (
        <VideoListContainer>
            {videos.items.map( (item) => (
                <VideoItem item={item} key={item.etag} />
            ))}
        </VideoListContainer>
    );

};

 export default VideoList;

 