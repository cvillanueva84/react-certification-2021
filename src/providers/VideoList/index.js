export { default, useAuth } from './Search.provider';
