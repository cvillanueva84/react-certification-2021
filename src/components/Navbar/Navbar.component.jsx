import React, { useState, useContext } from 'react';
import { Link } from 'react-router-dom';
import { FaHamburger, FaWindowClose, FaHome, FaUserNinja } from 'react-icons/fa';
import { IconContext } from 'react-icons/lib';
import './Navbar.styles.scss';
import { InputText } from './styledComponents';
import { VideoListContext } from '../../providers/VideoList/VideoList.provider';
import { useHistory } from 'react-router-dom';

// ##### Im thinking about refactoring this component into another component as I think it's kind of complex to read ####

const Navbar = () => {
  const [sidebar, setSidebar] = useState(false);
  const [inputValue, setInputValue] = useState('');
  const toogleSideBar = () => setSidebar(!sidebar);
  const { setSearch } = useContext(VideoListContext);
  const history = useHistory();

  const handleSubmit = (e) => {
    e.preventDefault();
    setSearch(inputValue);
    setInputValue('');
    history.push('/');
  };

  return (
    <>
      <IconContext.Provider value={{ color: '#fff', size: '3rem' }}>
        <nav className="navbar">
          <div className="navbar__left-section">
            <div className="menu-bars">
              <FaHamburger onClick={toogleSideBar} />
            </div>
            <form onSubmit={handleSubmit}>
              <InputText
                type="text"
                name="search"
                placeholder="search a video..."
                value={inputValue}
                onChange={(e) => setInputValue(e.target.value)}
              />
            </form>
          </div>

          <div className="navbar__right-section">
            <div className="toggle">
              <label className="switch">
                <input type="checkbox" />
                <span className="slider round"></span>
              </label>
            </div>
            <Link to="/">
              <FaUserNinja />
            </Link>
          </div>
        </nav>

        <nav className={sidebar ? 'nav-menu active' : 'nav-menu'}>
          <ul className="nav-menu-items" onClick={toogleSideBar}>
            <li
              className="navbar-toggle"
              style={{ cursor: 'pointer', marginLeft: '2rem' }}
            >
              <FaWindowClose />
            </li>
            <li className="nav-text">
              <Link to="/">
                <FaHome />
                <span>Home</span>
              </Link>
            </li>
          </ul>
        </nav>
      </IconContext.Provider>
    </>
  );
};

export default Navbar;
