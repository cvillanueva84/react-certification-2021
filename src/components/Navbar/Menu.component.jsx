import React, { useState } from 'react';
import MenuItem from '@material-ui/core/MenuItem';
import { withStyles } from '@material-ui/core/styles';
import Menu from '@material-ui/core/Menu';
import Login from '../Login';
import { useAuth } from '../../providers/Auth';

const MenuLogin = ({ anchor, toggleMenu }) => {
  const StyledMenu = withStyles({
    paper: {
      border: '1px solid #d3d4d5',
    },
  })((props) => (
    <Menu
      elevation={0}
      getContentAnchorEl={null}
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'center',
      }}
      transformOrigin={{
        vertical: 'top',
        horizontal: 'center',
      }}
      {...props}
    />
  ));

  const [loginForm, setLoginForm] = useState(false);

  const showLogin = () => {
    toggleMenu(null);
    setLoginForm(!loginForm);
  };

  const { authenticated, logout } = useAuth();

  const handleClose = () => {
    toggleMenu(null);
  };

  const handleLogout = () => {
    logout();
    toggleMenu(null);
  };

  return (
    <>
      <Login show={loginForm} toggleShow={showLogin} />
      <StyledMenu
        anchorEl={anchor}
        keepMounted
        open={Boolean(anchor)}
        onClose={handleClose}
      >
        {authenticated ? (
          <MenuItem onClick={handleLogout}>Log Out</MenuItem>
        ) : (
          <MenuItem onClick={showLogin}>Log In</MenuItem>
        )}
      </StyledMenu>
    </>
  );
};

export default MenuLogin;
