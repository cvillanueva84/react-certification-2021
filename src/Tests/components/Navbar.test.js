import React from 'react';
import { mount } from 'enzyme';
import Navbar from '../../components/Navbar/Navbar.component'
import { VideoListContext } from '../../providers/VideoList/VideoList.provider';
import { MemoryRouter } from 'react-router';

describe('Tests of the Navbar component', () => {
    
    const setSearch = jest.fn()
    let wrapper;
    beforeEach(() =>  {
        jest.clearAllMocks();
        wrapper = mount(
            <VideoListContext.Provider
                value={{
                    setSearch
                }}
            >
                <MemoryRouter>
                    <Navbar/>
                </MemoryRouter>
            </VideoListContext.Provider>)
    });

    test('should render the Navbar component correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
    
    test('should have an styled input tag', () => {
        const styledInput = wrapper.find('InputText')
        const value = 'React!'
        styledInput.simulate('change', {
            target: {
                value
            }
        })
        expect(styledInput.length).toBe(1)
    })
    test('should post info on submit', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect(setSearch).toHaveBeenCalled()
    })
    
    test('should call setSearch and clean the input text ', () => {
        const value = 'learn jest'
        wrapper.find('InputText').simulate('change', { target: {value}})
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect(setSearch).toHaveBeenCalled()
        expect(wrapper.find('InputText').prop('value')).toBe('')

    })
    
    
})

