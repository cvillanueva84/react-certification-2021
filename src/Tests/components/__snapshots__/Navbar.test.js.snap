// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of the Navbar component should render the Navbar component correctly 1`] = `
Array [
  <nav
    className="navbar"
  >
    <div
      className="navbar__left-section"
    >
      <div
        className="menu-bars"
      >
        <svg
          fill="currentColor"
          height="3rem"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "#fff",
            }
          }
          viewBox="0 0 512 512"
          width="3rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 256H48a48 48 0 0 0 0 96h416a48 48 0 0 0 0-96zm16 128H32a16 16 0 0 0-16 16v16a64 64 0 0 0 64 64h352a64 64 0 0 0 64-64v-16a16 16 0 0 0-16-16zM58.64 224h394.72c34.57 0 54.62-43.9 34.82-75.88C448 83.2 359.55 32.1 256 32c-103.54.1-192 51.2-232.18 116.11C4 180.09 24.07 224 58.64 224zM384 112a16 16 0 1 1-16 16 16 16 0 0 1 16-16zM256 80a16 16 0 1 1-16 16 16 16 0 0 1 16-16zm-128 32a16 16 0 1 1-16 16 16 16 0 0 1 16-16z"
            key="0"
          />
        </svg>
      </div>
      <form
        onSubmit={[Function]}
      >
        <input
          className="sc-bdnxRM gEZKDY"
          name="search"
          onChange={[Function]}
          placeholder="search a video..."
          type="text"
          value=""
        />
      </form>
    </div>
    <div
      className="navbar__right-section"
    >
      <div
        className="toggle"
      >
        <label
          className="switch"
        >
          <input
            type="checkbox"
          />
          <span
            className="slider round"
          />
        </label>
      </div>
      <a
        href="/"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="3rem"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "#fff",
            }
          }
          viewBox="0 0 448 512"
          width="3rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M325.4 289.2L224 390.6 122.6 289.2C54 295.3 0 352.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-70.2-54-127.1-122.6-133.2zM32 192c27.3 0 51.8-11.5 69.2-29.7 15.1 53.9 64 93.7 122.8 93.7 70.7 0 128-57.3 128-128S294.7 0 224 0c-50.4 0-93.6 29.4-114.5 71.8C92.1 47.8 64 32 32 32c0 33.4 17.1 62.8 43.1 80-26 17.2-43.1 46.6-43.1 80zm144-96h96c17.7 0 32 14.3 32 32H144c0-17.7 14.3-32 32-32z"
            key="0"
          />
        </svg>
      </a>
    </div>
  </nav>,
  <nav
    className="nav-menu"
  >
    <ul
      className="nav-menu-items"
      onClick={[Function]}
    >
      <li
        className="navbar-toggle"
        style={
          Object {
            "cursor": "pointer",
            "marginLeft": "2rem",
          }
        }
      >
        <svg
          fill="currentColor"
          height="3rem"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "#fff",
            }
          }
          viewBox="0 0 512 512"
          width="3rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-83.6 290.5c4.8 4.8 4.8 12.6 0 17.4l-40.5 40.5c-4.8 4.8-12.6 4.8-17.4 0L256 313.3l-66.5 67.1c-4.8 4.8-12.6 4.8-17.4 0l-40.5-40.5c-4.8-4.8-4.8-12.6 0-17.4l67.1-66.5-67.1-66.5c-4.8-4.8-4.8-12.6 0-17.4l40.5-40.5c4.8-4.8 12.6-4.8 17.4 0l66.5 67.1 66.5-67.1c4.8-4.8 12.6-4.8 17.4 0l40.5 40.5c4.8 4.8 4.8 12.6 0 17.4L313.3 256l67.1 66.5z"
            key="0"
          />
        </svg>
      </li>
      <li
        className="nav-text"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="3rem"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "#fff",
              }
            }
            viewBox="0 0 576 512"
            width="3rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
              key="0"
            />
          </svg>
          <span>
            Home
          </span>
        </a>
      </li>
    </ul>
  </nav>,
]
`;
